libtransmission/peer-mgr.h

```diff
@@ -316,17 +351,48 @@ class tr_peer_info
         return ret;
     }
 
-private:
-    [[nodiscard]] constexpr tr_address const& addr() const noexcept
-    {
-        return socket_address_.address();
-    }
+    // ---
 
-    [[nodiscard]] constexpr auto port() const noexcept
+    // merge a peer info objects into ourself
+    void merge(tr_peer_info const& that) noexcept
     {
-        return socket_address_.port();
+        connection_attempted_at_ = std::max(connection_attempted_at_, that.connection_attempted_at_);
+        connection_changed_at_ = std::max(connection_changed_at_, that.connection_changed_at_);
+        piece_data_at_ = std::max(piece_data_at_, that.piece_data_at_);
+
+        /* no need to merge blocklist since it gets updated elsewhere */
+
+        {
+            auto const conn_this = !is_connectable_ || *is_connectable_;
+            auto const conn_that = !that.is_connectable_ || *that.is_connectable_;
```

Comment:
```suggestion
            auto const conn_this = is_connectable_.value_or(true);
            auto const conn_that = that.is_connectable_.value_or(true);
```

Reply:
Right... I totally forgot about this API. 😅